e2etest/bgp_test.go

```diff
@@ -21,72 +21,107 @@ package e2e
 import (
 	"context"
 	"fmt"
-	"io/ioutil"
 	"net"
 	"strconv"
 	"time"
 
 	"go.universe.tf/metallb/e2etest/pkg/metrics"
 	"go.universe.tf/metallb/e2etest/pkg/routes"
 
+	"golang.org/x/sync/errgroup"
+
 	"github.com/onsi/ginkgo"
 	"github.com/onsi/ginkgo/extensions/table"
 	. "github.com/onsi/gomega"
 	dto "github.com/prometheus/client_model/go"
-	"go.universe.tf/metallb/e2etest/pkg/executor"
 	"go.universe.tf/metallb/e2etest/pkg/frr"
 	frrconfig "go.universe.tf/metallb/e2etest/pkg/frr/config"
+	frrcontainer "go.universe.tf/metallb/e2etest/pkg/frr/container"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	clientset "k8s.io/client-go/kubernetes"
 	"k8s.io/kubernetes/test/e2e/framework"
 	e2eservice "k8s.io/kubernetes/test/e2e/framework/service"
 )
 
-const frrContainer = "frr"
+const (
+	frrIBGP = "frr-iBGP"
+	frrEBGP = "frr-eBGP"
+	IBGPAsn = 64512
+	EBGPAsn = 64513
+)
 
 var (
-	frrContainerIPv4 string
-	frrContainerIPv6 string
+	frrContainers []*frrcontainer.Frr
 )
 
+type containerConfig struct {
+	name string
+	nc   frrconfig.NeighborConfig
+	rc   frrconfig.RouterConfig
+}
+
 var _ = ginkgo.Describe("BGP", func() {
 	f := framework.NewDefaultFramework("bgp")
 	var cs clientset.Interface
-
+	containersConf := []containerConfig{
+		{
+			name: frrIBGP,
+			nc: frrconfig.NeighborConfig{
+				ASN: IBGPAsn,
+			},
+			rc: frrconfig.RouterConfig{
+				ASN:     IBGPAsn,
+				BGPPort: 179,
+			},
+		},
+		{
+			name: frrEBGP,
+			nc: frrconfig.NeighborConfig{
+				ASN: EBGPAsn,
+			},
+			rc: frrconfig.RouterConfig{
+				ASN:     IBGPAsn,
+				BGPPort: 180,
```

Comment:
hmm this BGPPort is different than the one /neighbor don't we need to keep this at 0 as we have today and just change the /neighbor port if configured ?
`When port number is 0, that means do not listen bgp port.`

Reply:
This also relates to https://github.com/metallb/metallb/pull/1018#discussion_r740222489
that's the BGPPort that the external container will listen on so it is set here and reflected in the peer (c.RouterConfig.BGPPort) so the speaker would target the correct port.

Reply:
what if we change the port later `neighbor 1.1.1.1 port 200` and what will happen if we have 
`neighbor 1.1.1.1 port 100`
`nieghbor 1.1.1.2 port 110`
which one will be used for BGPPort ?

Reply:
ok that is clear now thanks